### ISSUE TITLE
Mixed workload operation types must be one of {'put','get','delete','updatemeta','head'}, but got

### ISSUE THREAD

**Issue by scaleoutsean** [s3tester/s3tester#24]
Using current release on Ubuntu 18.04 I get this incomplete message about my mixedWorkload file:
```
$ ./s3tester `
-bucket one001 `
-concurrency 1 `
-endpoint "https://111.111.111.111:1443" `
-ratelimit 4 `
-requests 100 `
-size 8888 `
-workload "w.json"

2020/11/02 12:17:37 Mixed workload operation types must be one of {'put','get','delete','updatemeta','head'}, but got
```

That's all it says.
w.json looks good:

```
$ cat w.json

{
  "mixedWorkload": [
    {
      "operation": "put",
      "ratio": 50
    },
    {
      "operationType": "get",
      "ratio": 50
    }
  ]
}
```

Executed without `-workload`, it works fine.

**Comment by scaleoutsean** [comment]
Also tried with the invalid JSON from README.md, no luck.
```
$ cat w.json
'{'mixedWorkload':
    [{'operation':'put','ratio':25},
    {'operationType':'get','ratio':25},
    {'operationType':'updatemeta','ratio':25},
    {'operationType':'delete','ratio':25}]}'

$ ./s3tester -bucket one001 -concurrency 1 -endpoint "https://10.128.59.215:1443" -ratelimit 1 -requests 100 -size 2000 -workload "w.json"
panic: runtime error: integer divide by zero

$ go version
go version go1.15 linux/amd64
goroutine 1 [running]:
main.calcStats(0xc000263678, 0x1, 0xdb221)
        /home/sean/go/src/github.com/s3tester/s3tester/s3tester.go:460 +0x185
main.setupResultStat(0xc000263678)
        /home/sean/go/src/github.com/s3tester/s3tester/s3tester.go:448 +0x49
main.processTestResult(0xc000263678, 0x1, 0x7d0, 0xc000091aa0, 0x1, 0x1, 0x9db2e0, 0x3, 0x7ffc0ed43680, 0x6, ...)
        /home/sean/go/src/github.com/s3tester/s3tester/s3tester.go:435 +0x74
main.runtest(0x1, 0x7d0, 0xc000091aa0, 0x1, 0x1, 0x9db2e0, 0x3, 0x7ffc0ed43680, 0x6, 0x9ddca8, ...)
        /home/sean/go/src/github.com/s3tester/s3tester/s3tester.go:187 +0x1f8
main.main()
        /home/sean/go/src/github.com/s3tester/s3tester/s3tester.go:576 +0x9d8

**Comment by ohaensel** [comment]
i get exactly the same behavior (s3tester 2.1.0

**Comment by vmore** [comment]
Thanks for the info. I think you can try using "operationType" exclusively. Drop the first mention of "operation".

**Comment by scaleoutsean** [comment]
@vmore are you saying this would work? I don't have access to S3 right now. 

```
{
  "mixedWorkload": [
    {
      "operationType": "put",
      "ratio": 50
    },
    {
      "operationType": "get",
      "ratio": 50
    }
  ]
}
```

**Comment by ohaensel** [comment]
Thanks Vlad,
that made the trick.

Now the mixed workload is executed.
I noticed that the command output refers to the first operation mentioned (here “put”) although using a slightly modified workload profile
[root@cbc-hadoop01 SG]# cat mixed1.json
{
   "mixedWorkload":[
      {
         "operationType":"put",
         "ratio":25
      },
      {
         "operationType":"get",
         "ratio":15
      },
      {
         "operationType":"head",
         "ratio":50
      },
      {
         "operationType":"delete",
         "ratio":10
      }

   ]
}

        --- Total Results ---
Operation: put
Concurrency: 1024
Total number of requests: 200000
Failed requests: 0
Total elapsed time: 2m0.954621641s
Average request time: 14.76135ms
Minimum request time: 1.96ms
Maximum request time: 635.91ms
Nominal requests/s: 69370.3
Actual requests/s: 1653.5
Content throughput: 0.056769 MB/s
Average Object Size: 36
Response Time Percentiles
50     :   4.58 ms
75     :   18.97 ms
90     :   40.97 ms
95     :   54.36 ms

I suggest correcting the readme as it requires to use operation is first parameter.

Kind regards
Oliver

**Comment by vmore** [comment]
This is now fixed by #23 . Thanks everyone.
